canal/sync.go

```diff
@@ -63,11 +59,16 @@ func (c *Canal) runSyncBinlog() error {
 		if ev.Header.LogPos == 0 {
 			switch e := ev.Event.(type) {
 			case *replication.RotateEvent:
-				fakeRotateLogName = string(e.NextLogName)
+				fakeRotateLogName := string(e.NextLogName)
 				c.cfg.Logger.Infof("received fake rotate event, next log name is %s", e.NextLogName)
+				if fakeRotateLogName != c.master.Position().Name {
+					c.cfg.Logger.Info("log name changed, the fake rotate event will be handled as a real roate event")
```

Comment:
NIT: `...as a real rotate event`? (and in the name of PR as well)

Reply:
Done. Thx.